### PR TITLE
profanity: enable python and gpg support, parallel builds and enforcement of enabled features

### PR DESCRIPTION
**Pull request by Moredread**
###### Motivation for this change

Enabling python and gpg support, parallel builds and enforcement of enabled features.

If wanted, I can split up the PR or squash commits as needed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Moredread** [comment]
cc @devhell

File: pkgs/applications/networking/instant-messengers/profanity/default.nix [hunk 1]
```diff
@@ -21,13 +25,25 @@ stdenv.mkDerivation rec {
...
+
   buildInputs = [
     readline libuuid libmesode
     glib openssl expat ncurses libotr curl
   ] ++ optionals autoAwaySupport [ libXScrnSaver libX11 ]
     ++ optionals notifySupport   [ libnotify gdk_pixbuf ]
-    ++ optionals traySupport     [ gnome2.gtk ];
+    ++ optionals traySupport     [ gnome2.gtk ]
+    ++ optionals gpgSupport      [ gpgme ]
+    ++ optionals pythonSupport   [ python ];
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/profanity/default.nix]: What python version is preferred here? Archlinux builds against python3

Comment (Moredread) [reply pkgs/applications/networking/instant-messengers/profanity/default.nix]: This is set at https://github.com/Moredread/nixpkgs/blob/ebd3a4622453665df5d600ab4a69c9d23060b88c/pkgs/top-level/all-packages.nix#L16067-L16068 which is python 2. I intended to use python 3, I'll change it.

**Comment by FRidh** [comment]
> python support

Does that mean Python bindings? In that case, can you add the package also to `python-packages.nix`. See https://github.com/NixOS/nixpkgs/pull/30167 as an example.

**Comment by Moredread** [comment]
@FRidh I think it's only for plugin support, i.e. interacting _from_ python is not possible as far as I can see. I renamed the option so it is more clear.

**Comment by Moredread** [comment]
Mhh, it seems that it doesn't compile with the tray icon feature flag (so the last commit actually catches the problem, until now it just wasn't compiled in). Can someone on MacOS have a look whether tray support works on that platform or needs to be disabled?

**Comment by Moredread** [comment]
I renamed the gpgSupport option to pgpOption to match the usage in the package

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -16063,9 +16063,12 @@ with pkgs;
...
-    notifySupport   = config.profanity.notifySupport   or true;
-    traySupport     = config.profanity.traySupport     or true;
-    autoAwaySupport = config.profanity.autoAwaySupport or true;
+    notifySupport       = config.profanity.notifySupport         or true;
+    traySupport         = config.profanity.traySupport           or true;
+    autoAwaySupport     = config.profanity.autoAwaySupport       or true;
+    pgpSupport          = config.profanity.pgpSupport            or true;
+    pythonPluginSupport = config.profanity.pythonPluginSupport   or true;
+    python              = config.profanity.pythonPackage         or python3;
   };
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: Can you get rid of these config options. If one wants to change an option one can just override the expression.

Comment (Moredread) [reply pkgs/top-level/all-packages.nix]: All of them, or only those I've added?

**Pull request merged**